seml/start.py

```diff
@@ -472,26 +569,44 @@ def print_commands(db_collection_name, unobserved, post_mortem, num_exps, filter
     logging.info(" ".join(config))
 
     logging.info("\nCommand for running locally with post-mortem debugging:")
-    configs = start_jobs(db_collection_name, slurm=False,
-                         unobserved=True, post_mortem=True,
-                         num_exps=1, filter_dict=filter_dict, dry_run=True)
-    exe, _, config = configs[0]
+    exe, config = get_command_from_exp(exps_list[0], collection.name,
+                                       verbose=logging.root.level <= logging.VERBOSE,
+                                       unobserved=unobserved, post_mortem=True)
     logging.info(f"python {exe} with {' '.join(config)}")
 
     logging.info("\n********** All raw commands **********")
     logging.root.setLevel(orig_level)
-    configs = start_jobs(db_collection_name, slurm=False,
-                         unobserved=unobserved, post_mortem=post_mortem,
-                         num_exps=num_exps, filter_dict=filter_dict, dry_run=True)
-    for (exe, _, config) in configs:
+    start_experiments(collection, local=True,
+                      unobserved=unobserved, post_mortem=post_mortem,
+                      num_exps=num_exps, filter_dict=filter_dict, dry_run=True)
+    for exp in exps_list:
+        exe, config = get_command_from_exp(exp, collection.name,
+                                           unobserved=unobserved, post_mortem=post_mortem)
         logging.info(f"python {exe} with {' '.join(config)}")
 
 
 def start_experiments(db_collection_name, local, sacred_id, batch_id, filter_dict,
                       num_exps, unobserved, post_mortem, debug, dry_run,
-                      output_to_console):
+                      output_to_console, no_file_output, local_steal_slurm,
+                      no_worker, set_to_pending=True, worker_gpus=None, worker_cpus=None, worker_environment_vars=None):
+
     use_slurm = not local
-    output_to_file = not output_to_console
+    output_to_file = not no_file_output
+    launch_worker = not no_worker
+
+    if filter_dict is None:
+        filter_dict = {}
+
+    if use_slurm and not set_to_pending:
```

Comment:
What's the use case for set_to_pending=False?

Reply:
See comment above: `set_to_pending=False` captures the case that we only want to launch a local worker. The worker only takes jobs from the `PENDING` queue but does not set `STAGED` experiments to `PENDING`.